// nekobot-hentai/index.js
module.exports = {
    Hentai: {
      hentai: require("./func/hentai.js"),
      pussy: require("./func/pussy.js"),
      gah: require("./func/gah.js"),
      coffee: require("./func/coffee.js"),
      food: require("./func/food.js"),
      paizuri: require("./func/paizuri.js"),
      tentacle: require("./func/tentacle.js"),
      yaoi: require("./func/yaoi.js"),
      hmidriff: require("./func/hmidriff.js"),
      pgif: require("./func/pgif.js"),
      holo: require("./func/holo.js"),
      gonewild: require("./func/gonewild.js"),
      kanna: require("./func/kanna.js"),
      kemonomimi: require("./func/kemonomimi.js"),
      hkitsune: require("./func/hkitsune.js"),
      ass: require("./func/ass.js"),
      hass: require("./func/hass.js"),
      boobs: require("./func/boobs.js"),
      hboobs: require("./func/hboobs.js"),
      thigh: require("./func/thigh.js"),
      hthigh: require("./func/hthigh.js"),
      anal: require("./func/anal.js"),
      hanal: require("./func/hanal.js"),
    }
}
